autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -205,312 +204,62 @@ def get_model_level(self, model_name):
-        if n_repeat_start == 0:
-            time_start = time.time()
-            model_names_trained = self.train_multi_initial(X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val, models=models, kfolds=kfolds, n_repeats=n_repeats_initial, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune,
-                                                           stack_name=stack_name, level=level, time_limit=time_limit)
-            n_repeat_start = n_repeats_initial
-            if time_limit is not None:
-                time_limit = time_limit - (time.time() - time_start)
-        else:
-            model_names_trained = models
-        if (n_repeats > 1) and self.bagged_mode and (n_repeat_start < n_repeats):
-            model_names_trained = self.train_multi_repeats(X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val, models=model_names_trained,
-                                                           kfolds=kfolds, n_repeats=n_repeats, n_repeat_start=n_repeat_start, stack_name=stack_name, level=level, time_limit=time_limit, time_limit_total_level=time_limit_total_level)
-        return model_names_trained
-
-    def train_multi_and_ensemble(self, X_train, y_train, X_val, y_val, models: List[AbstractModel], hyperparameter_tune=True, feature_prune=False, time_limit=None):
-        if self.save_data and not self.is_data_saved:
-            self.save_X_train(X_train)
-            self.save_y_train(y_train)
-            if X_val is not None:
-                self.save_X_val(X_val)
-                if y_val is not None:
-                    self.save_y_val(y_val)
-            self.is_data_saved = True
-
-        self.num_rows_train = len(X_train)
-        if X_val is not None:
-            self.num_rows_train += len(X_val)
-        self.num_cols_train = len(list(X_train.columns))
-        self.train_multi_levels(X_train, y_train, X_val, y_val, models=models, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, level_start=0, level_end=self.stack_ensemble_levels, time_limit=time_limit)
-        if len(self.get_model_names_all()) == 0:
-            raise ValueError('AutoGluon did not successfully train any models')
+    # TODO: make models accept dictionary of level -> list of models for more control of future fit calls.
+    # TODO: Enable HPO on levels > 0
+    # TODO: Enable feature prune on levels > 0
+    # TODO: Remove name_suffix, hacked in
+    # TODO: Enable easier re-mapping of trained models -> hyperparameters input (They don't share a key since name can change)
+    def train_multi_levels(self, X_train, y_train, X_val, y_val, hyperparameters: dict = None, base_model_names: List[str] = None, hyperparameter_tune=False, feature_prune=False, core_kwargs: dict = None, level_start=0, level_end=0, stack_name_core='core', stack_name_aux='aux1', time_limit=None, name_suffix='') -> List[str]:
+        """
+        Trains a multi-layer stack ensemble using the input data on the hyperparameters dict input.
```

Comment:
maybe clarify hyperparameters dict is used to determine the models in each stack layer?

Reply:
added